wal.go

```diff
@@ -68,7 +68,7 @@ func (b *azureSecretBackend) rollbackAppWAL(ctx context.Context, req *logical.Re
 	// found, so no special handling is needed for that case. If we don't succeed within
 	// maxWALAge (e.g. client creds have changed and the delete will never succeed),
 	// unconditionally remove the WAL.
-	if err := client.deleteApp(ctx, entry.AppObjID, false); err != nil {
+	if err := client.deleteApp(ctx, entry.AppObjID, true); err != nil {
```

Comment:
@dpowley do you recall why we had specifically set this to `false` in #104?

Reply:
Looks like it was done to preserve the original behavior. The change proposed here makes sense to avoid dangling apps.